python/teams_ai/turn_state/turn_state_manager.py

```diff
@@ -3,31 +3,81 @@
 Licensed under the MIT License.
 """
 
-import abc
-from typing import TypeVar, Generic, TypedDict, Union
+from typing import TypeVar, Generic, Optional, List
 from botbuilder.core import TurnContext, Storage
+
 from .turn_state_entry import TurnStateEntry
+from .turn_state import TurnState
+
+TurnStateT = TypeVar("TurnStateT", bound=TurnState)
 
 
-class TurnState(TypedDict):
-    "base interface defining a collection of turn state scopes"
+class TurnStateManager(Generic[TurnStateT]):
```

Comment:
In js SDK, it's called `DefaultTurnStateManager`. I personally prefer to keep the class name aligned between different languages (could have different naming style per language) to avoid user confusions. @Stevenic what's your opinion to this?

BTW, in dotnet SDK it's called `TurnStateManager`. If we decide to keep the class name aligned, we need to see whether the js SDK requires update, or python/dotnet SDK requires update.